### PR TITLE
Release/v0.2.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: 0.2.2

## Release details

# AEA

- Adds support for registry flags on `autonomy scaffold fsm` command
- Adds support for scaffolding
  - Dialogues
  - Payloads
  - Round tests
  - Behaviour tests
- Adds support for specifying version for runtime images
- Removes the need for `NESTED_FIELDS_ALLOWED_TO_UPDATE` from service config class
- Uses local file for service registry ABI rather than fetching from staging server
- Replaces the usage of staging chain with locally deployed chain

# Packages
- Adds support for broadcasting APY estimates to a backend server.
- Replaces API keys with environment variable placeholders
- Ports tests for packages to their respective package folders

# Plugins
- Introduces `aea-test-autonomy` plugin

# Chores
- Bumps `tendermint` to `v0.35.7`
- Fixes tendermint logging issues
- Updates skaffold config for building runtime images for agents
- Bumps autonolas registries sub module to latest

# Tests
- Replaces the usage of staging chain with local registry deployments
- Fixes flaky `test_fetch_behaviour_non_indexed_block` test
- Fixes flaky registry tests
- Adds tests for `autonomy scaffold fsm`

# Docs
- Restructures documentation to introduce new index
- Adds tutorials on creating a service using an existing agent
- Updates quick start documentation
- Adds overview for service development process
- Adds docs for fsm scaffolding tool

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side), from `develop`
- [x] I've updated the dependencies versions to the latest, wherever is possible.
- [x] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [x] I built the documentation and updated it with the latest changes
- [x] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ Below we describe the additional manual steps required to upgrade between differ
 
 # Open Autonomy
 
+## `v0.2.1.post1` to `v0.2.2`
+
+No backwards incompatible changes
```
Comment (angrybayblade) [inline docs/upgrading.md:10]: AFAIC!

File: HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,46 @@
 # Release History - `open-autonomy`
 
+# 0.2.0 (2022-08-09)
```
Comment (DavidMinarsch) [inline HISTORY.md:3]: Wrong version: `0.2.2`

File: HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+
+# AEA
+ 
+- Adds support for registry flags on `autonomy scaffold fsm` command
+- Adds support for scaffolding
+  - Dialogues
+  - Payloads
+  - Round tests
+  - Behaviour tests
```
Comment (Karrenbelt) [inline HISTORY.md:12]: also scaffolding tests for: handlers, models, payloads, dialogues

File: HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,46 @@
...
+- Uses local file for service registry ABI rather than fetching from staging server
+- Replaces the usage of staging chain with locally deployed chain
+
+# Packages
+- Adds support for broadcasting APY estimates to a backend server.
+- Replaces API keys with environment variable placeholders
+- Ports tests for packages to their respective package folders
+
+# Plugins
+- Introduces `aea-test-autonomy` plugin
```
Comment (DavidMinarsch) [inline HISTORY.md:24]: Should also start at `0.2.2`

File: HISTORY.md [hunk 5]
```diff
@@ -1,5 +1,46 @@
...
+
+# Chores
+- Bumps `tendermint` to `v0.35.7`
```
Comment (DavidMinarsch) [inline HISTORY.md:27]: No that's not correct

File: docs/upgrading.md (hunk 1 above)
Comment (DavidMinarsch) [reply docs/upgrading.md:10]: I think only in tests...and so yeah hopefully you're right

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: HISTORY.md (hunk 3 above)
Comment (angrybayblade) [reply HISTORY.md:12]: Just `Tests` then?

**Pull request merged**